### DISCUSSION TITLE
自定义明细表单元格报错 - Cannot read properties of undefined (reading 'theme')

### DISCUSSION THREAD

**Discussion by May-yfm** [antvis/S2#2827] [General]
**s2版本**是：2.0.0-next.10

**问题**：自定义明细表的对齐方式，报错：Cannot read properties of undefined (reading 'theme') 是什么原因？？

参照官网案例写的：https://s2.antv.antgroup.com/zh/examples/custom/custom-cell/#custom-table-cell

在官网案例中，点击右上角【在CodeSandbox中打开】，也是报错了，链接：https://codesandbox.io/s/pthgzp

这样怎么改啊？急急急！

**Comment by lijinke666** [comment]
2.0.0-next.10 版本有点太老了, 请优先考虑升级版本, 你这个版本需要这样写

```diff
dataCell: (viewMeta, spreadsheet) => {
- return new CustomDataCell(viewMeta, spreadsheet)
+ return new CustomDataCell(viewMeta, viewMeta?.spreadsheet)
}
```

> 点击右上角【在CodeSandbox中打开】，也是报错了

codesandbox 默认安装的是 latest, 对应的是 1.x 版本, 应该使用 `next`, 版本不对肯定就报错了.

![image](https://github.com/user-attachments/assets/dbe45bf6-66cd-4149-8b0b-b697bf497f37)

**Reply by May-yfm** [reply]
好的嘞，非常感谢~
我直接 $ npm install @antv/s2@next --save 就给我安装了next.10
基础版本，现在最新是到 2.0.0-next.25版本了吗？那我需要更新一下
